up-l1/someip.adoc

```diff
@@ -0,0 +1,265 @@
+
+`Instance ID` *MAY* be encoded in the upper 16 bits of the 32 bit `ue_id`.
+If the upper 16 bits are `0x0000` (a reserved instance value), `Instance ID` is assumed to be `1`.
+
+NOTE: Explicitly setting `Instance ID` to `0x0001` *SHOULD* be avoided as it increases the message size.
+
+a| `ue_version_major` a| `Interface Version`
+a| `Interface Version` field of *SOME/IP Header*. `[PRS_SOMEIP_00053]`.
+
+Maps to `Major Version` for the `Service ID`. If set, it *SHOULD* be passed to underlying *SOME/IP* library to override it's default _Service Major Version_
+
+a| `resource_id` a| `Method ID`/`Event ID` | Identifier of the method/event placed in the
+lower 16 bits of the *SOME/IP Header* `Message ID`. `[PRS_SOMEIP_00034]`
+
+|===
+
+*NOTES:*
+
+    * `Service ID` and `Method ID` are defined in *SOME/IP Header format*, specified by AUTOSAR `[PRS_SOMEIP_00030]`.
+    * `Event ID` is an alias for `Method ID`, but used for events. `[PRS_SOMEIP_00245]`
+    * `Instance ID` is defined in *SOME/IP-SD Service Entry Type*, specified by AUTOSAR `[PRS_SOMEIPSD_00268]`.
+    * `Instance ID` is not present in *SOME/IP Header*, but is needed for _Service Discovery_ and events and *MUST* be provided for *SOME/IP* libraries (e.g. `vsomeip`). `[PRS_SOMEIP_00162]`
+    * Additional mappings *MAY* be required by the underlying *SOME/IP* library, e.g. Service Major/Minor version. Such values *MAY* be pre-configured
+
+==== UUIDs
+
+link:../basics/uuid.adoc[uProtocol UUID] specifications create a unique identifier for each message along with timestamp information.
+
+The ID is used for correlate between request and response as well. *SOME/IP* instead defines the `Request ID` as 16 bit `Client ID` + 16 bit `Session ID` (that is incremented). `[PRS_SOMEIP_00046]`
+
+When messages are converted *SOME/IP* to/from *uProtocol*, care must be taken to ensure that the *SOME/IP* ID and uProtocol UUID are properly mapped, especially when corelating a request to a response.
+
+* Generated *SOME/IP* Events *MUST* set the 16 bit `Client ID` to 0 per `[PRS_SOMEIP_00925]`
+* Generated *SOME/IP* Responses *MUST* auto-populate the `Request ID` cached from the request message, into the response message and then flush the entry in the cache.
+
+Further details of the usage of IDs for the various message types can be found in the next section.
+
+==== Message Type
+
+<<umessagetype-mapping>> table below maps of uProtocol messages to `[PRS_SOMEIP_00055]` *SOME/IP* message types.
+
+.UMessageType Mapping
+[umessagetype-mapping, cols="1,1,2"]
+|===
+| UMessageType | SOME/IP Type | Details
+
+a| `UMESSAGE_TYPE_PUBLISH` a| `NOTIFICATION` | Publish
+
+a| `UMESSAGE_TYPE_REQUEST` a| `REQUEST` | Requests
+
+a| `UMESSAGE_TYPE_RESPONSE` a| `RESPONSE` or `ERROR` | Responses or error has occurred while attempting to deliver the message or a service
+has thrown an exception
+
+a| `UMESSAGE_TYPE_NOTIFICATION` a| `NOTIFICATION` | Notification
```

Comment:
Per our chat, this is a different type of message that is a request with no response correct?

Reply:
`UMESSAGE_TYPE_NOTIFICATION` was previously mapped to `REQUEST_NO_RESPONSE`, which seems wrong.

I can't tell whether to use `UMESSAGE_TYPE_PUBLISH` or `UMESSAGE_TYPE_NOTIFICATION` for incoming SOME/IP events. e.g. if we have a subscriber to the specific `service/instance/event`, it should be a  `UMESSAGE_TYPE_NOTIFICATION`,  if no subscribers are found send it as `UMESSAGE_TYPE_PUBLISH`?